src/reflection.ts

```diff
@@ -247,11 +247,19 @@ export class Parser {
     const lambdas: Record<string, (t: Table) => any> = {};
     const schema = this.getType(typeIndex);
     const numFields = schema.fieldsLength();
-    for (let ii = 0; ii < numFields; ++ii) {
-      const field = schema.fields(ii);
+
+    // Sort fields by ID so the resulting object is built with fields in this order
+    const sortedFields: reflection.Field[] = [];
+    for (let i = 0; i < numFields; ++i) {
+      const field = schema.fields(i);
       if (field === null) {
-        throw new Error("Malformed schema: field at index " + ii + " not populated.");
+        throw new Error(`Malformed schema: field at index ${i} not populated.`);
       }
+      sortedFields.push(field);
+    }
+    sortedFields.sort((a, b) => a.id() - b.id());
```

Comment:
This logic makes me question why we keep lambdas as a Record rather than a list of deserializers since the lambdas are never exposed and exist only to iterate on field names later:
https://github.com/foxglove/flatbuffers_reflection/blob/main/src/reflection.ts#L346-L348

I think you could instead build up a list of deserializers that includes the field name, id, deserializer function and then sort that on the _id_. Then in the returned closure we don't need to do field name lookups and can iterate a list.